core/src/components/checkbox/checkbox.ionic.scss

```
@@ -44,94 +56,340 @@
 }
 
 // Checkbox Disabled
 // --------------------------------------------------
+:host(.checkbox-disabled) {
+  pointer-events: none;
+}
+
 // disabled, indeterminate checkbox
 :host(.checkbox-disabled.checkbox-indeterminate) .checkbox-icon {
-  /* TODO(FW-6183): Use design token variables */
-  border-width: 0;
+  border-width: globals.$ionic-border-size-0;
 
-  background-color: #{$background-color-step-600};
+  background-color: #{globals.$ionic-color-base-white};
 }
 
 // disabled, unchecked checkbox
 :host(.checkbox-disabled) .checkbox-icon {
-  /* TODO(FW-6183): Use design token variables */
-  border-color: #c9c9c9;
+  border-color: globals.$ionic-color-neutral-800;
 
-  background-color: #f5f5f5; // mix of #f5f5f5 with 60% #FFF
+  background-color: globals.$ionic-state-disabled;
 }
 
 // disabled, checked checkbox
 :host(.checkbox-disabled.checkbox-checked) .checkbox-icon {
-  border-width: 0;
+  border-width: globals.$ionic-border-size-0;
+
+  background-color: globals.$ionic-color-primary-base;
 
-  background-color: $background-color-step-100;
+  opacity: 0.6;
 }
 
 // Checkbox Hover
 // --------------------------------------------------------
 @media (any-hover: hover) {
   :host(:hover) .checkbox-icon {
-    /* TODO(FW-6183): Use design token variables */
-    background-color: #ececec; // mix of 'white', '#121212', 0.08, 'rgb'
+    background-color: globals.$ionic-color-neutral-100;
   }
 
   :host(:hover.checkbox-checked) .checkbox-icon,
   :host(:hover.checkbox-indeterminate) .checkbox-icon {
-    /* TODO(FW-6183): Use design token variables */
-    background-color: #1061da; // mix of '#1068eb', '#121212', 0.08, 'rgb'
+    background-color: globals.$ionic-color-primary-800;
   }
 }
 
 // Checkbox Focus
 // --------------------------------------------------
 // Only show the focus ring when the checkbox is focused and not disabled
 :host(.ion-focused:not(.checkbox-disabled)) .checkbox-icon {
-  outline: var(--focus-ring-width) solid var(--focus-ring-color);
+  outline: var(--focus-ring-width) globals.$ionic-border-style-solid var(--focus-ring-color);
   outline-offset: var(--focus-ring-offset);
 }
 
 // Checkbox: Active
 // --------------------------------------------------------
 :host(.ion-activated) .checkbox-icon {
-  /* TODO(FW-6183): Use design token variables */
-  background-color: #e3e3e3; // mix of 'white', '#121212', 0.12, 'rgb'
+  background-color: globals.$ionic-color-neutral-200;
 }
 
 :host(.ion-activated.checkbox-checked) .checkbox-icon,
 :host(.ion-activated.checkbox-indeterminate) .checkbox-icon {
-  /* TODO(FW-6183): Use design token variables */
-  background-color: #105ed1; // mix of '#1068eb', '#121212', 0.12, 'rgb'
+  background-color: globals.$ionic-color-primary-900;
 }
 
 // Ionic Design Checkbox Shapes
 // --------------------------------------------------
 :host(.checkbox-shape-soft) {
-  --border-radius: #{$checkbox-ionic-border-radius};
+  --border-radius: #{globals.$ionic-border-radius-100};
 }
 
```

This is where we would override the border radius for soft and small.

```suggestion

:host(.checkbox-size-small.checkbox-shape-soft) {
  --border-radius: #{globals.$ionic-border-radius-050};
}

```